### PR TITLE
User guide quantum kernel

### PR DESCRIPTION
**Pull request by JanSchnabel**

### CODE REVIEW

File: docs/user_guide/kernel_methods.rst [hunk 1]
```diff
@@ -6,26 +6,301 @@
...
 
-**To be done!**
+Due to the inner product, the kernel function formally computes the distance betweed data points
+:math:`x` and :math:`y` and thus effectively reduces to the illustrative interpretation of a 
+similarity measure between data points.
 
-High-Level methods that employ quantum kernels:
------------------------------------------------
+The key point of Quantum Kernel methods is that they can be fundamentally formulated as a classical 
+kernel method whose kernel is computed by a quantum computer. By Using a quantum computer for the
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:28]: using

File: docs/user_guide/kernel_methods.rst [hunk 2]
```diff
@@ -6,26 +6,301 @@
...
+*superposition* and *entanglement* - a fact which holds out the prospect of designing machine 
+learning models that are able deal with complex problems that are out of reach for conventional 
+machine learning methods.
 
-:class:`ml.QSVR`
-:class:`ml.QSVC`
-:class:`ml.QGPR`
-:class:`ml.QGPC`
-:class:`ml.QKRR`
+Quantum Kernel methods work analougsly to their classical counterparts, with data embedded into
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:34]: analogously

File: docs/user_guide/kernel_methods.rst [hunk 3]
```diff
@@ -6,26 +6,301 @@
...
 
+:math:`\rho(x,\boldsymbol{\theta})=\ket{\psi(x,\boldsymbol{\theta})}\bra{\psi(x,\boldsymbol{\theta})}`
 
-Training of quantum kernels:
-----------------------------
+enriched with the *Hilbert-Schmidt inner product* as the feature space of a quantum model and 
+state [1]. Thus, in quantum computing, access to the Hilbert space of quantum states is given by 
+measurements. In contrast to conventional kernel methods, the central task for quantum kernel 
+methods is to find sufficiently well suited quantum feature maps for a given data set, while in 
+classical kernel methods one never only treats feature maps implicitly by selecting a proper
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:48]: den Satz verstehe ich nicht

File: docs/user_guide/kernel_methods.rst [hunk 4]
```diff
@@ -6,26 +6,301 @@
...
-:class:`matrix.ProjectedQuantumKernel`
+with :math:`U_{\boldsymbol{\theta}}(x)` a parametrized quantum circuit for data enconding applied 
+to the initial qubit state :math:`\ket{0}`. At this, we can consider the space of complex matrices
 
+:math:`\rho(x,\boldsymbol{\theta})=\ket{\psi(x,\boldsymbol{\theta})}\bra{\psi(x,\boldsymbol{\theta})}`
 
-Training of quantum kernels:
-----------------------------
+enriched with the *Hilbert-Schmidt inner product* as the feature space of a quantum model and 
+state [1]. Thus, in quantum computing, access to the Hilbert space of quantum states is given by
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:45]: state -> states?

Der Übergang zu density matrices ist ein bisschen holprig.

File: docs/user_guide/kernel_methods.rst [hunk 5]
```diff
@@ -6,26 +6,301 @@
...
+
+.. currentmodule:: squlearn.kernel.ml
+
+High-Level methods that employ quantum kernels
+----------------------------------------------
+In general, kernel methods refer to a collection of pattern analysis algorithms that use kernel 
+functions to operate in a high-dimensional feature space. Probably, the most famous representative
+of these kernel-based algorithms is *Support Vector Machines (SVMs)*. Kernel methods are most 
+commonly used in a supervised learning framework for either classification or regression tasks. 
+But, there are use-cases for kernel-based unsupervised algorithms too; however, these are (currently)
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:60]: Not necessary to point out that we don't cover that in my opinion.

File: docs/user_guide/kernel_methods.rst [hunk 6]
```diff
@@ -6,26 +6,301 @@
...
+not covered by sQUlearn. 
+
+In the NISQ era (no access to fast linear algebra algorithms such as HHL), the basic notion of 
+quantum kernel methods is to merely compute the kernel matrix with a quantum computer and 
+subsequently pass it to an conventional kernel algorithms. For this task, sQUlearn provides a 
+convenient workflow by either wrapping the corresponding scikit-learn estimators or by 
+independetly implementing them analougsly, adapted to the needs of quantum kernel methods. 
+sQUlearn offers SVMs for both classifciation (QSVC) and regression (QSVR) tasks, Gaussian
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:68]: can we link to the methods in the documentation?

File: docs/user_guide/kernel_methods.rst [hunk 7]
```diff
@@ -6,26 +6,301 @@
...
+convenient workflow by either wrapping the corresponding scikit-learn estimators or by 
+independetly implementing them analougsly, adapted to the needs of quantum kernel methods. 
+sQUlearn offers SVMs for both classifciation (QSVC) and regression (QSVR) tasks, Gaussian 
+Processes (GPs) for both classification (QGPC) and regression (QGPR) tasks as well as a 
+quantum kernel ridge regression routine (QKRR).
+
+Classification
+##############
+
+In terms of classification tasks, sQUlearn provides:
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:75]: tasks -> algorithms

File: docs/user_guide/kernel_methods.rst [hunk 8]
```diff
@@ -6,26 +6,301 @@
...
+   QSVC
+   QGPC
+
+We refer to the documentations and examples of the respective methods for in-depth information
+and user guidelines.
+
+Regression
+##########
+
+In terms of regression tasks, sQUlearn provides:
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:89]: tasks -> algorithms

File: docs/user_guide/kernel_methods.rst [hunk 9]
```diff
@@ -6,26 +6,301 @@
...
+        executor=Executor('statevector_simulator')
+    )
+
+When evaluating kernels on a real backend, sQUlearn provides an option for mitigating FQKs for
+with respect to depolarizing noise using the approach discussed in Ref. [2]. The respective
+mitigation technique uses the fact that ideally (train-train or test-test) kernel matrices 
+should consist exclusively of ones along the diagonal - a property which is by construction
+fulfilled by PQKs. For FQKs one can attempt to restore this property using the 
+:code:`mit_depol_noise` argument which can be either set to *msplit* or *mmean* as defined
+in Ref. [2]. By default this option is set to None.
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:158]: :code:`None`?

File: docs/user_guide/kernel_methods.rst [hunk 10]
```diff
@@ -6,26 +6,301 @@
...
+
+
+Methods to evaluate quantum kernels
+-----------------------------------
+sQUlearn provides two methods to evaluate quantum kernels: Fidelity Quantum kernels (FQKs) and 
+Projected Quantum kernels (PQKs), which also represent the standard approaches to quantum kernel 
+methods in the literature.
+
+Central to both approaches is the embedding of data into the quantum Hilbert space by using
+quantum feature maps, which are nothing but parameterized quantum circuits. At this, the
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:109]: Maybe simpler:

...by using quantum feature maps which are nothing but encoding quantum circuits. These can optionally be parametrized for optimally adjusting the quantum kernel...

File: docs/user_guide/kernel_methods.rst [hunk 11]
```diff
@@ -6,26 +6,301 @@
...
+as a solution. These work by projecting the quantum states to an approximate classical
+representation by using, e.g., reduced physical observables. 
+
+The default PQK implementation of sQUlearn is one of the simplest forms of PQKs. They work by 
+measuring the one-particle reduced density matrix (1-RDM) on all qubits for the encoded state
+and define the kernel as (RBF-inspired)
+
+:math:`k^{PQ}(x,x^\prime)=\exp\left(-\gamma\sum_k\sum_{P\in\lbrace X,Y,Z\rbrace}\left[\mathrm{tr}(P\rho(x,\boldsymbol{\theta})_k) - \mathrm{tr}(P\rho(x^\prime,\boldsymbol{\theta})_k)\right]^2\right)`
+
+where :math:`\rho(x,\boldsymbol{\theta})_k` refers to the 1-RDM,
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:176]: maybe writing `\rho(x,\boldsymbol{\theta})_k=Tr_k.... helps the illustration?

File: docs/user_guide/kernel_methods.rst [hunk 12]
```diff
@@ -6,26 +6,301 @@
...
+        executor=Executor('statevector_simulator'),
+        measurement='XYZ',
+        outer_kernel='gaussian'
+    )
+
+Moreover, the QNNs can be evaluated with respect to different measurement operators, where in
+addition to the default setting - :code:`measurement='XYZ'` - one can specify :code:`measurement='X'`, 
+:code:`measurement='Y'` and :code:`measurement='Z'` for one-qubit measurements with respect to only 
+one Pauli operator. Beyond that, one can also specify an operator or a list of operators, see the 
+respective examples in :class:`ProjectedQuantumKernel` or the "Operators for expectation values"
```
Comment (rothma) [inline docs/user_guide/kernel_methods.rst:211]: can we link to the user guide?

**Comment by rothma** [comment]
Thanks for putting in so much effort, I think it's really good. I added some (minor) remarks directly in the file.

File: docs/user_guide/kernel_methods.rst (hunk 3 above)
Comment (JanSchnabel) [reply docs/user_guide/kernel_methods.rst:48]: Der ist auch sehr umständlich gebaut und führt wahrscheinlich nur zu Verwirrungen - habe ihn gelöscht.

**Review by rothma — APPROVED** [review]

**Pull request merged**